DynamicIntArray.cpp

```diff
@@ -0,0 +1,90 @@
+#include "DynamicIntArray.h"
+#include <iostream>
+
+DynamicIntArray::DynamicIntArray(): m_size(0), m_array(new int[0] {}) {};
+
+DynamicIntArray::DynamicIntArray(std::size_t size): m_size(size), m_array(new int[size] {}) {};
+
+DynamicIntArray::DynamicIntArray(const DynamicIntArray& other) {
+
+	m_size = other.m_size;
+	m_array = new int[m_size];
+	
+	for (int i = 0; i < m_size; i++) {
+		m_array[i] = other.m_array[i];
+	}
+}
+
+DynamicIntArray& DynamicIntArray::operator=(const DynamicIntArray& other) {
+
+	delete[] m_array; 
+
+	m_size = other.m_size;
+	m_array = new int[m_size];
+
+	for (int i = 0; i < m_size; i++) {
+		m_array[i] = other.m_array[i];
+	}
+
+	return *this;
+}
+
+DynamicIntArray::~DynamicIntArray() {
+	delete[] m_array;
+}
+
+int& DynamicIntArray::operator[](std::size_t index) {
+	return m_array[index];
+}
+
+void DynamicIntArray::setSize(std::size_t newSize) {
+	std::size_t loopSize = (m_size < newSize) ? m_size : newSize;
+
+	int* tmp = new int[newSize] {}; 
+
+	for (int i = 0; i < loopSize; i++) {
+		tmp[i] = m_array[i];
+	}
+
+	delete[] m_array;
+	m_array = new int[newSize] {};
+
+	for (int i = 0; i < loopSize; i++) {
+		m_array[i] = tmp[i];
+	}
```

Comment:
This second loop is redundant.
It's enough to reassign pointers
m_array = tmp;

Reply:
if I do this way, I get an error: 

![image](https://github.com/iarkov/r_d_cpp_gamedev/assets/31201969/96b2c13f-c809-4f16-aae5-66be2d594253)
![image](https://github.com/iarkov/r_d_cpp_gamedev/assets/31201969/c514a79e-5532-4d40-b841-501283fd7b89)